src/main/java/org/opensearch/neuralsearch/processor/util/ProcessorUtils.java

```diff
@@ -413,4 +422,93 @@ public static int getMaxTokenCount(
+            return ((MatchQueryBuilder) queryBuilder).value().toString();
+        }
+
+        if (queryBuilder instanceof NeuralQueryBuilder) {
+            return ((NeuralQueryBuilder) queryBuilder).queryText();
+        }
+
+        if (queryBuilder instanceof TermQueryBuilder) {
+            return ((TermQueryBuilder) queryBuilder).value().toString();
+        }
+
+        if (queryBuilder instanceof BoolQueryBuilder boolQuery) {
+            StringBuilder text = new StringBuilder();
+
+            for (QueryBuilder clause : boolQuery.must()) {
+                String clauseText = extractQueryTextFromBuilder(clause);
+                if (!clauseText.isEmpty()) {
+                    if (!text.isEmpty()) text.append(" ");
+                    text.append(clauseText);
+                }
+            }
+
+            for (QueryBuilder clause : boolQuery.should()) {
+                String clauseText = extractQueryTextFromBuilder(clause);
+                if (!clauseText.isEmpty()) {
+                    if (!text.isEmpty()) text.append(" ");
+                    text.append(clauseText);
+                }
+            }
+
+            return text.toString();
+        }
+
+        if (queryBuilder instanceof HybridQueryBuilder hybridQuery) {
+            StringBuilder text = new StringBuilder();
+
+            for (org.opensearch.index.query.QueryBuilder subQuery : hybridQuery.queries()) {
+                String subText = extractQueryTextFromBuilder(subQuery);
+                if (!subText.isEmpty()) {
+                    if (!text.isEmpty()) text.append(" ");
+                    text.append(subText);
+                }
+            }
+
+            return text.toString();
+        }
+
+        if (queryBuilder instanceof NestedQueryBuilder nestedQuery) {
+            return extractQueryTextFromBuilder(nestedQuery.query());
+        }
+
+        // Throw exception for unsupported query types
+        throw new IllegalArgumentException(
+            String.format(Locale.ROOT, "Query type %s is not supported for semantic highlighting", queryBuilder.getClass().getSimpleName())
+        );
+    }
+
+    /**
+     * Update SearchResponse with additional processing time
+     */
+    public static SearchResponse updateResponseTookTime(SearchResponse response, long additionalTimeMs) {
+        try {
+            long originalTook = response.getTook().millis();
+            long newTook = originalTook + additionalTimeMs;
+            return new org.opensearch.action.search.SearchResponse(
+                response.getInternalResponse(),
+                response.getScrollId(),
+                response.getTotalShards(),
+                response.getSuccessfulShards(),
+                response.getSkippedShards(),
+                newTook,
+                response.getShardFailures(),
+                response.getClusters(),
+                response.pointInTimeId()
+            );
+        } catch (Exception e) {
```

Comment:
Few questions.
1. Why do we need to catch exception here? I don't see any code which might throw an exception?
2. If exception handling is necessary, do we have unit test testing this case?
3. If exception handling is necessary, shouldn't we also log error message?

Reply:
updated